src/main/java/io/github/lottetreg/httpserver/Router.java

```diff
@@ -0,0 +1,94 @@
+package io.github.lottetreg.httpserver;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Router {
+  private List<Routable> routes;
+  private Outable out;
+
+  Router(List<Routable> routes) {
+    this.routes = routes;
+    this.out = new Out();
+  }
+
+  Router(List<Routable> routes, Outable out) {
+    this.routes = routes;
+    this.out = out;
+  }
+
+  public HTTPResponse route(HTTPRequest request) {
+    try {
+      Routable route = findMatchingRoute(request);
+      HTTPResponse response = route.getResponse(request);
+
+      if(request.getMethod().equals("OPTIONS")) {
+        response.addHeader("Allow", getAllowedMethods(request));
+      }
```

Comment:
I think I'll want to perform this conditionally changing the response somewhere else, but I'm leaving it here for now until I have more cases